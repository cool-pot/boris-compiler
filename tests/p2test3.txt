*** simple calc with variables and control structures
global a=0;
global b=1;

*** simplest foreach I can think of
foreach c in 1 .. 10 do print c; end for

*** more complicated stuff
foreach c in 1 .. 10 do
  if c==(c/2)*2 then *** even c
     print 2;
  else *** odd
     print 1;
  end if
end for