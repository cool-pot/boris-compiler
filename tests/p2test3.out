> Hello from parserTester for boris compiler!
Start parse tree visualization:

[ROOT_INPUT]
    [SDD_LIST]
        [DECL_AS_SDD]
            [GLOBAL_DECL]
                [PLACEHOLDER-261]global
                [ID]a
                [PLACEHOLDER-293]=
                [SINGLE_INT_AS_EXPR]
                    [INT]0
                [PLACEHOLDER-290];
        [SDD_LIST]
            [DECL_AS_SDD]
                [GLOBAL_DECL]
                    [PLACEHOLDER-261]global
                    [ID]b
                    [PLACEHOLDER-293]=
                    [SINGLE_INT_AS_EXPR]
                        [INT]1
                    [PLACEHOLDER-290];
            [SDD_LIST]
                [STATEMENT_AS_SDD]
                    [FOREACH_RANGE_STATEMENT]
                        [PLACEHOLDER-270]foreach
                        [ID]c
                        [PLACEHOLDER-272]in
                        [RANGE]
                            [SINGLE_INT_AS_EXPR]
                                [INT]1
                            [PLACEHOLDER-275]..
                            [SINGLE_INT_AS_EXPR]
                                [INT]10
                        [PLACEHOLDER-265]do
                        [STATEMENT_LIST]
                            [PRINT_STATEMENT]
                                [PLACEHOLDER-274]print
                                [SINGLE_ID_AS_EXPR]
                                    [ID]c
                                [PLACEHOLDER-290];
                            [EMPTY_NODE]
                        [PLACEHOLDER-263]end
                        [PLACEHOLDER-271]for
                [SDD_LIST]
                    [STATEMENT_AS_SDD]
                        [FOREACH_RANGE_STATEMENT]
                            [PLACEHOLDER-270]foreach
                            [ID]c
                            [PLACEHOLDER-272]in
                            [RANGE]
                                [SINGLE_INT_AS_EXPR]
                                    [INT]1
                                [PLACEHOLDER-275]..
                                [SINGLE_INT_AS_EXPR]
                                    [INT]10
                            [PLACEHOLDER-265]do
                            [STATEMENT_LIST]
                                [IF_ELSE_STATEMENT]
                                    [PLACEHOLDER-266]if
                                    [BOOLEXPR]
                                        [SINGLE_ID_AS_EXPR]
                                            [ID]c
                                        [PLACEHOLDER-282]==
                                        [EXPR_MULT_EXPR]
                                            [LPAR_EXPR_LPAR]
                                                [PLACEHOLDER-291](
                                                [EXPR_DIV_EXPR]
                                                    [SINGLE_ID_AS_EXPR]
                                                        [ID]c
                                                    [PLACEHOLDER-287]/
                                                    [SINGLE_INT_AS_EXPR]
                                                        [INT]2
                                                [PLACEHOLDER-292])
                                            [PLACEHOLDER-286]*
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]2
                                    [PLACEHOLDER-267]then
                                    [STATEMENT_LIST]
                                        [PRINT_STATEMENT]
                                            [PLACEHOLDER-274]print
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]2
                                            [PLACEHOLDER-290];
                                        [EMPTY_NODE]
                                    [EMPTY_NODE]
                                    [ELSE_SENTENCE]
                                        [PLACEHOLDER-269]else
                                        [STATEMENT_LIST]
                                            [PRINT_STATEMENT]
                                                [PLACEHOLDER-274]print
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]1
                                                [PLACEHOLDER-290];
                                            [EMPTY_NODE]
                                    [PLACEHOLDER-263]end
                                    [PLACEHOLDER-266]if
                                [EMPTY_NODE]
                            [PLACEHOLDER-263]end
                            [PLACEHOLDER-271]for
                    [EMPTY_NODE]
